src/main/java/core/basesyntax/AbstractFigure.java

```diff
@@ -0,0 +1,19 @@
+package core.basesyntax;
+
+public abstract class AbstractFigure implements Figure {
+    protected String color;
+
+    public AbstractFigure(String color) {
+        this.color = color;
+    }
+
+    public String getColor() {
+        return color;
+    }
+
+    @Override
+    public void draw() {
+        System.out.println("Figure Color: " + color);
+    }
```

Comment:
Since AbstractFigure is an abstract class, it might be more appropriate to declare the draw method as abstract, allowing each subclass to implement its own version of the draw method. This would provide more flexibility and adhere to the principle of abstraction.